engine/src/main/java/org/terasology/config/ServerInfo.java

```diff
@@ -32,58 +30,103 @@
         // for serialization purposes
     }
 
+
+    /**
+     * ServerInfo constructor
+     * Example parameters to pass: ServerInfo("My New Terasology Server", "my.address", 12345);
+     *
+     * @param name    Sets the name of the server.
+     * @param address Sets the address (URL) of the server.
+     * @param port    Sets the port of the server.
+     */
     public ServerInfo(String name, String address, int port) {
         setName(name);
         setAddress(address);
         setPort(port);
     }
 
+    /**
+     * @return Returns the name of the server
+     */
     public String getName() {
         return name;
     }
 
+    /**
+     * Precondition: The name of the server is not null or empty
+     * @param name Sets the name of the server
+     */
     public void setName(String name) {
         Preconditions.checkArgument(name != null && !name.isEmpty(), "Server name must not be null or empty");
         this.name = name;
     }
 
+    /**
+     * @return Returns the address (URL) of the server
+     */
     public String getAddress() {
         return address;
     }
 
+    /**
+     * Precondition: The address (URL) is not null or empty
+     * @param address Sets the address
+     */
     public void setAddress(String address) {
         Preconditions.checkArgument(address != null && !address.isEmpty(), "Server address must not be null or empty");
         this.address = address;
     }
 
+    /**
+     * @return Returns the port
+     */
     public int getPort() {
         return port;
     }
 
+    /**
+     * Precondition: The port ranges from 0 to 65535 inclusive
+     * @param port Sets the port
```

Comment:
As rzats wrote before, no need to detail all Preconditions, but in this case you can simply write:

@param port an integer representing the port number, in the [0-65535] range, inclusive.